tasks/main.yml

```diff
@@ -29,6 +44,58 @@
 - name: Flush handlers
   meta: flush_handlers
 
-- name: Perform join operation
-  shell: echo {{ ad_integration_password }} | realm join {{ ad_integration_realm }}
-  when: not ansible_check_mode
+## Execute other roles if applicable
+- name: Add AD server to existing network connection for DNS
+  include_role:
+   name: fedora.linux_system_roles.network
+  vars:
+    network_connections:
+      - name: "{{ ad_integration_dns_connection_name }}"
+        interface_name: ""
+        type: "{{ ad_integration_dns_connection_type }}"
+        ip:
+          dns: "{{ ad_integration_dns_server }}"
+    network_allow_restart: yes
+  when: ad_integration_manage_dns | bool
+
+- name: Manage timesync
+  include_role:
+   name: fedora.linux_system_roles.timesync
+  vars:
+    timesync_ntp_servers:
+      - hostname:  "{{ ad_integration_timesync_source }}"
+        iburst: yes
+  when: ad_integration_manage_timesync | bool
+
+- name: Manage crypto policies
+  include_role:
+   name: fedora.linux_system_roles.crypto_policies
+  vars:
+    crypto_policies_policy: "DEFAULT:AD-SUPPORT"
+  when:
+    - ad_integration_manage_crypto_policies | bool
+    # Fedora and RHEL8+
+    - (ansible_distribution == "Fedora" or (ansible_distribution in ['CentOS', 'RedHat'] and ansible_distribution_version is version('8', '>=')))
+
+## RHEL9 uses the AD-SUPPORT-LEGACY policy for RC4, otherwise AD-SUPPORT allows it
+- name: Enable crypto policy allowing RC4 encryption
+  include_role:
+   name: fedora.linux_system_roles.crypto_policies
+  vars:
+    crypto_policies_policy: "DEFAULT:AD-SUPPORT-LEGACY"
+  when:
+    - ad_integration_allow_rc4_crypto | bool
+    - ansible_distribution in ['CentOS', 'RedHat']
+    - ansible_distribution_version is version('9', '>=')
+
+- name: Join to a specific Domain Controller
+  shell: echo {{ ad_integration_password }} | realm join -U {{ ad_integration_user }} --membership-software {{ ad_integration_membership_software }} {{ ad_integration_join_to_dc }}
```

Comment:
```yaml
  shell: |
    set -euo pipefail
    echo {{ ad_integration_password | quote }} | realm join -U \
      {{ ad_integration_user | quote }} --membership-software \
      {{ ad_integration_membership_software | quote }} \
      {{ ad_integration_join_to_dc | quote }}
```

Reply:
Also - is this idempotent?  That is, if already joined to a specific Domain Controller, will this report `changed: true` or `changed: false`?  Is there some way to know if a machine is already joined and report `changed: false` if so?

Reply:
Attempting to run the role a second time after already joined shows this error as realm reports `realm: Already joined to this domain`. How should this be addressed?

~~~
TASK [linux-system-roles.template : Perform discovery-based realm join operation] **************************************************************************************************************************************
task path: /home/justin/github/ad_integration/tests/roles/linux-system-roles.template/tasks/main.yml:112
fatal: [10.0.138.38]: FAILED! => {
    "changed": true,
    "cmd": "echo Secret123 | realm join -U Administrator --membership-software adcli domain.com",
    "delta": "0:00:00.026805",
    "end": "2022-09-20 10:23:20.162984",
    "rc": 1,
    "start": "2022-09-20 10:23:20.136179"
}

STDERR:

realm: Already joined to this domain
Please check
    https://red.ht/support_rhel_ad 
to get help for common issues.


MSG:

non-zero return code
~~~

Reply:
use `register`, `changed_when`, and `failed_when`.  For example, if the exit code `1` means already joined, and other exit codes mean some sort of "real" error condition:
```yaml
  ...
  register: __realm_join_output
  failed_when: __realm_join_output.rc not in [0, 1]
  changed_when: __realm_join_output.rc == 0
```
or, if there is not such an exit code, you'll have to parse the stderr output:
```yaml
  ...
  register: __realm_join_output
  failed_when:
    - __realm_join_output is failed
    - not __realm_join_output.stderr is search("Already joined to this domain")
  changed_when: not __realm_join_output.stderr is search("Already joined to this domain")
```
or something like that